lib/matplotlib/contour.py

```diff
@@ -969,19 +969,29 @@ def _autolev(self, N):
         Select contour levels to span the data.
 
         The target number of levels, *N*, is used only when the
-        scale is not log and default locator is used.
+        locator is not set and the scale is log or the default
+        locator is used.
 
         We need two more levels for filled contours than for
         line contours, because for the latter we need to specify
         the lower and upper boundary of each range. For example,
         a single contour boundary, say at z = 0, requires only
         one contour line, but two filled regions, and therefore
         three levels to provide boundaries for both regions.
+
+        Parameters
+        ----------
+        N : int | None
         """
         if self.locator is None:
             if self.logscale:
-                self.locator = ticker.LogLocator()
+                if N is None:
+                    self.locator = ticker.LogLocator()
+                else:
+                    self.locator = ticker.LogLocator(numticks=N)
```

Comment:
Actually, `numticks=None` is the default.
```suggestion
                self.locator = ticker.LogLocator(numticks=N)  # N may be None
```